src/componetnts/pagination/Pagination.tsx

```diff
@@ -0,0 +1,89 @@
+import styles from './pagination.module.scss'
+import classNames from 'classnames/bind';
+import { getArrayNumbersFromNumber, splitItemsOnArraysInArray } from '../../utils/pagination';
+
+interface PaginationProps {
+  activePage: number;
+  totalItems: number;
+  perPage: number;
+  withActions?: boolean;
+  classes?: {
+    btn?: string;
+    activeBtn?: string;
+  };
+  onChangePage: (newPage: number) => void;
+};
+
+const cx = classNames.bind(styles);
+const defaultBtnStyles = {
+  btn: 'button',
+  activeBtn: 'activeButton'
+}
+
+const Pagination = ({
+  activePage,
+  totalItems,
+  perPage,
+  withActions = true,
+  classes = defaultBtnStyles,
+  onChangePage,
+}: PaginationProps) => {
+  const quantityOfPages: number = Math.ceil(totalItems / perPage);
+  const quantityButtonsShow: number = 3;
+  const arrayDigits: number[] = getArrayNumbersFromNumber(quantityOfPages);
+  const arrayItems: number[][] = splitItemsOnArraysInArray(arrayDigits, quantityButtonsShow);
+  const activePageIndex: number = arrayItems.findIndex((array) => array.includes(activePage));
+
+  const handlePreviousPage = (): void => {
+    if (activePage !== 1) {
```

Comment:
instead of making this check every click just disable the button if you are on the end